src/services/array-functions.js

```diff
@@ -1,57 +1,84 @@
 
 //in the function map, create a new array and store in a variable
-//loop theArray and call the fnc for each thing in the array, 
+//loop theArray and call the fnc for each thing in the array,
 //     passing in the item from the current loop into the call to fnc
 //add the returned value from fnc to the new array
 //return the new array
 export function map(theArray, fnc){
-
+  let newArray = [];
+  for (var i = 0; i < theArray.length; i++){
+    newArray.push(fnc(theArray[i]));
+ }
+ return newArray;
 }
 
 //create a new array
-//loop theArray and call the fnc for each thing in the array, 
+//loop theArray and call the fnc for each thing in the array,
 //     passing in the item from the current loop
 //fnc will return true or false, if true add the item to the new array else do not
 //return the new array
 export function filter(theArray, fnc){
-
+  let newArr = [];
+  for (let i = 0; i < theArray.length; i++) {
+    if(fnc(theArray[i])) {
+      newArr.push(theArray[i]);
+    }
+  }
+  return newArr;
 }
 
 
-//loop theArray and call the fnc for each thing in the array, 
+
+//loop theArray and call the fnc for each thing in the array,
 //     passing in the item from the current loop
-//fnc will return true or false, if true return the item 
+//fnc will return true or false, if true return the item
 //return null
 export function find(theArray, fnc){
-
-}
+  let finalArr = [];
+   for (let i = 0; i < theArray.length; i++) {
+     if(fnc(theArray[i])) {
+       finalArr.push(theArray[i]);
```

Comment:
double check this code. We don't need to add anything to an array in the find function.